src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs

```diff
@@ -1787,6 +1790,302 @@ public void TestSqlCommandSequentialAccessCodePaths(string connection, CommandBe
+            Assert.Equal(rowsAffected, numberOfRows);
+
+            using (SqlConnection sqlConnection = new SqlConnection(connection))
+            {
+                sqlConnection.Open();
+
+                using (SqlCommand sqlCommand = new SqlCommand($@"SELECT * FROM {tableName} WHERE FirstName = @FirstName AND CustomerId = @CustomerId",
+                    sqlConnection,
+                    transaction: null,
+                    columnEncryptionSetting: SqlCommandColumnEncryptionSetting.Enabled))
+                {
+                    sqlCommand.Parameters.AddWithValue(@"FirstName", values[1]);
+                    sqlCommand.Parameters.AddWithValue(@"CustomerId", values[0]);
+
+                    TaskCompletionSource<object> completion = new TaskCompletionSource<object>();
+                    IAsyncResult asyncResult = sqlCommand.BeginExecuteReader(new AsyncCallback(EndExecuteReaderAsyncCallBack),
+                        stateObject: new TestAsyncCallBackStateObject(sqlCommand, expectedRowsAffected: commandbehavior == CommandBehavior.SingleRow ? 1 : numberOfRows,
+                        completion: completion, expectedValues: values, commandBehavior: commandbehavior),
+                        behavior: commandbehavior);
+                    Assert.True(asyncResult != null, "asyncResult should not be null.");
+                }
+            }
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
+        [ClassData(typeof(AEConnectionStringProviderWithExecutionMethod))]
+        public void TestSqlCommandCancel(string connection, string value, int number)
+        {
+            string executeMethod = value;
+            Assert.True(!string.IsNullOrWhiteSpace(executeMethod), @"executeMethod should not be null or empty");
+
+            int numberofCancelCalls = number;
+            Assert.True(numberofCancelCalls >= 0, "numberofCancelCalls should be >=0.");
+
+            IList<object> values = GetValues(dataHint: 58);
+            Assert.True(values != null && values.Count >= 3, @"values should not be null and count should be >= 3.");
+
+            int numberOfRows = 300;
+            int rowsAffected = InsertRows(tableName: tableName, numberofRows: numberOfRows, values: values, connection: connection);
+            Assert.True(rowsAffected == numberOfRows, "number of rows affected is unexpected.");
+
+            using (SqlConnection sqlConnection = new SqlConnection(connection))
+            {
+                sqlConnection.Open();
+
+                using (SqlCommand sqlCommand = new SqlCommand($@"SELECT * FROM [{tableName}] WHERE FirstName = @FirstName AND CustomerId = @CustomerId",
+                    sqlConnection,
+                    transaction: null,
+                    columnEncryptionSetting: SqlCommandColumnEncryptionSetting.Enabled))
+                {
+                    sqlCommand.CommandTimeout = 0;
```

Comment:
Why this is set to `0` here?

Reply:
Fixed. Setting commandTimeOut to 0 means it is set to wait indefinitly. I have taken that out as it was not needed.